### PR TITLE
url prefix resolution

### LINKED ISSUE #28: Investigate intermittent test failure for successful PUT on specification file
### Description of the issue

In the setup-ci branch (which included cypress tests) a delay was added to force the test to wait for an async await to resolve before continuing with the test https://github.com/fiaisis/admin-portal/pull/18/commits/769b9c2126a8e78cdb4c9b426b426cf208e3bb13

In a previous investigation of running the failing test 30 times, test #9 and #24 failed. This should be investigated (i.e. run for a sufficient numbe of cases (100?) if any failures occur, then investigate it more thoroughly and find a solution

Issue resolution: 
There needed to be an explicit check of the contents of the JSON (by accessing the code editor contents) before moving onto the next steps, the failing tests were when the JSON wasn't awaited so attempting to PUT it resulted in no JSON (unexpected end of JSON)

### LINKED ISSUE #42: [staging] admin-portal prefix missing during fetch calls
### Description of the issue

The admin-portal is making requests to the wrong url in the staging environment. 
Despite all the previous routing url fixes this one was overlooked. 

Both requests(GET and PUT) are being sent to `/api/instrument/ALF/specification` with the response being a 403 forbidden error `{"detail":"Invalid authentication credentials"}`

Prepending the correct "admin-portal" prefix to the PUT request (via network requests) i.e.`/admin-portal/api/instrument/ALF/specification` returns a 500 internal server error `error: Object { }`.

To fix the api route and investigate the 500 error.

### PR DESCRIPTION
**Pull request by HumzahJavid**
Closes #42 (along the way it closed #28 , #46 and #49)

## Description
This PR will add:
- Local FIA API API KEY value, to enable testing locally
- A data-cy variable to support the tests below in locating the JSON specification
- A new viewspec cypress test and is reused in other tests (fixes intermittent issues #28 )
- Update styling to make page responsive (fixes styling issue #46 )
- Change logic for applying theme to alert using statusCode rather than statusText (fixes alert issue #49)
- Add BASE_URL (as env variable, literal in next.config.mjs and as a constants.ts file).
- Use NODE_ENV as a production flag in above BASE_URL to dynamically change url.
- Update tests with BASE_URL 
- Update README with running cypress against dev environment
- Add a CI_TEST env variable to be used for correcting cypress test BASE_URL logic on CI

### CODE REVIEW

**Comment by HumzahJavid** [comment]
After the previous commit, the test broke locally, whilst fixing them an intermittent timeout error was identified. 

If the cypress suite is run immediately after starting a local dev server, the `Opens specification page` test fails

image

But a fix was identified by adding a timeout the calling line { timeout: 6000 }

File: cypress/e2e/spec.cy.ts [hunk 1]
```diff
@@ -12,7 +14,54 @@ describe('E2E Test', () => {
...
-    cy.url().should('include', '/specification/MARI');
+    cy.url({ timeout: 6000 }).should('include', '/specification/MARI');
+  });
+});
+
+describe('E2E Test', () => {
+  it('View specification', () => {
+    const specification = { now: false, call: false, term: false };
+    const specificationString = JSON.stringify(specification);
+    console.log(specificationString);
```
Comment (keiranjprice101) [inline cypress/e2e/spec.cy.ts:25]: Do we still need this log?

File: cypress/e2e/spec.cy.ts [hunk 2]
```diff
@@ -79,23 +128,26 @@ describe('E2E Test', () => {
 
 describe('E2E Test', () => {
   it('A successful specification update', () => {
+    const specification = { now: false, call: false, term: false };
+    const specificationString = JSON.stringify(specification);
+    console.log(specificationString);
```
Comment (keiranjprice101) [inline cypress/e2e/spec.cy.ts:133]: And here

**Review by keiranjprice101 — COMMENTED** [review]
Nice work. Just 2 points

**Review by keiranjprice101 — APPROVED** [review]

**Pull request merged**